### ISSUE TITLE
Calls.iter should have same API as Calls.list

### ISSUE THREAD

**Issue by buddylindsey** [twilio/twilio-python#163]
I have needed to pull in all calls, or a large amount, from the api a couple of times for data syncing issues and verification issues. It works reasonably well for all of them.

Unfortunately, this is not necessarily easy if you want a subset using the `iter`, and don't want to use an excess of logic for paging with `list`.

List Resource provides an `iter` for pulling in data, but I have to "re-implement" part of `Calls.list` to get it to work correctly for giving a range of dates. It would be nice to have an override for `Calls.iter` to be similar to `Calls.list`.

Specifically for my purposes was.

```python
client = TwilioRestClient('', '')
kwargs = {}
kwargs['StartTime>'] = 'some date'
kwargs['EndTime<'] = 'some other date'

calls = client.calls.iter(**kwargs)
```
This is a fairly straight forward implementation after it was done, and made sense, but took me a couple of hours to figure out. After trying a whole bunch of other things. It wasn't intuitive to learn.

My thought is something like: (without testing it)
```python
def iter(self, from_=None, ended_after=None,
         ended_before=None, ended=None, started_before=None,
         started_after=None, started=None, **kwargs):

    kwargs["from"] = from_
    kwargs["StartTime<"] = started_before
    kwargs["StartTime>"] = started_after
    kwargs["StartTime"] = parse_date(started)
    kwargs["EndTime<"] = ended_before
    kwargs["EndTime>"] = ended_after
    kwargs["EndTime"] = parse_date(ended)

    super(Calls, self).iter(kwargs)
```

**Comment by buddylindsey** [comment]
Was this closed just because or was it implemented...? Was asked to provide this feedback, but am assuming now it isn't needed?

**Comment by carlosdp** [comment]
@buddylindsey sorry, I should have added an explanation to this one (I've been cleaning out old issues today). This was actually fixed by 840ea843 .

**Comment by buddylindsey** [comment]
@carlosdp awesome. Thank you. :)
